crates/broker/src/submitter.rs

```diff
@@ -58,15 +58,25 @@ pub enum SubmitterErr {
     UnexpectedErr(#[from] anyhow::Error),
 }
 
+impl_coded_debug!(SubmitterErr);
+
 impl CodedError for SubmitterErr {
     fn code(&self) -> &str {
         match self {
             SubmitterErr::UnexpectedErr(_) => "[B-SUB-500]",
             SubmitterErr::AllRequestsExpiredBeforeSubmission(_) => "[B-SUB-001]",
             SubmitterErr::SomeRequestsExpiredBeforeSubmission(_) => "[B-SUB-005]",
             SubmitterErr::MarketError(_) => "[B-SUB-002]",
-            SubmitterErr::BatchSubmissionFailed(_) => "[B-SUB-003]",
-            SubmitterErr::TxnConfirmationError(_) => "[B-SUB-004]",
+            SubmitterErr::BatchSubmissionFailed(submitter_errs) => {
+                // If all errors are txn confirmation errors, then we use a separate code to enable custom alerting.
+                if submitter_errs.iter().all(|e| matches!(e, SubmitterErr::TxnConfirmationError(_)))
+                {
+                    "[B-SUB-003]"
+                } else {
+                    "[B-SUB-004]"
+                }
+            }
```

Comment:
How hard would it be to move this to when the error is created? It seems a bit off to do this in the `Debug` impl

Reply:
Handled https://github.com/boundless-xyz/boundless/pull/774/commits/864e05c5c02483fc44a7a1967332c36ffbcd20c1